### ISSUE TITLE
Questioning config in mixins vs dedicated config files

### ISSUE THREAD

**Issue by saviomuc** [xing/hops#663]
## This is my intent (choose one)

- [ ] I want to report a bug
- [x] I want to request a feature or change
- [ ] I want to provide or change a feature

## The problem
Recently I had to remove the hops postcss plugin because it would hard code post css and some post css plugins within the config: https://github.com/xing/hops/blob/master/packages/postcss/mixin.core.js
I think this approach is valid if you own the code and know how to extend it (which is kinda hard in hops, when you don't understand the way the mixins work).

However imo the postcss preset is very opinionated, might not fit each case and prevents overriding via config file.
 It would be much easier to adopt and use if it would just load the  `postcss-loader` and leave the rest to the user's `postcss.config.js`.

The same idea should apply for everything else imo. If there is a config file it should be first to be used. I.e. for `.babelrc`. Next for instance does the same: 
https://github.com/zeit/next.js/#customizing-babel-config 

This is so much easier for me because it lets me see the complete information about the project's babel dependencies in one place. Currently I need to do some mental mapping whenever I want to understand which dependencies are used.

To make this work for hops it would be extremely helpful to document which hops presets use which babel config / plugin / preset. Another way would be to declare all babel-dependencies as peer dependencies. This would at least help to understand which babel-dependencies are really used and installed.

I understand that other projects like `Create React App` do something very similar like hops, BUT there is basically a one click eject button, which exposes all config. This is not working in hops.

## Proposed solution
- I'd be happy to create a PR for postcss which dramatically reduces its functionality. Maybe the preset should just be renamed to `hops-cssmodules`, then the settings would make sense.
- For the babel issue it's a RFC

**Comment by dmbch** [comment]
Hey @saviomuc, thank you for your thoughtful input! I think you are addressing a couple of rather interesting topics in your remarks. Let me try to explain our current reasoning with regards to some of them:

**PostCSS**

You are completely right, of course: that module is quite opinionated. Just as with Babel, we primarily support *future standard* language features to ensure maximum stability and maintainability. Oh, and we tried to not break too much existing code when we switched to `postcss-preset-env` after `postcss-cssnext` was deprecated. We should probably find a way to deal with that compatibility stuff and make it optional or add it in a downstream (i.e. XING-internal) package.

**Separate Configs**

We decided early on to paradigmatically only support *the hops way* of configuring the individual tools we integrate. We do so to ensure the reusability and shareability of those configs (w/o copy-paste).  Additionally, we need multiple (aligned) configs for different environments and targets (development/production, node/browser) - and we want to provide clear/deterministic extension points for specific mixins. Both would be rather impossible if we were to support arbitrary configs.

**Documentation**

This is an area hops is currently a bit lacking, especially in comparison to the tools you mentioned above. We are working on it - and I absolutely support your idea of thoroughly documenting the babel/postcss/etc-configs we provide.
